PCSamples/Graphics/SimpleHDR_PC/DeviceResources.cpp

```diff
@@ -550,11 +567,52 @@ void DeviceResources::UpdateColorSpace()
 #if defined(NTDDI_WIN10_RS2)
     if (m_swapChain)
     {
-        ComPtr<IDXGIOutput> output;
-        if (SUCCEEDED(m_swapChain->GetContainingOutput(output.GetAddressOf())))
+        if (!m_dxgiFactory->IsCurrent())
+        {
+            // Output information is cached on the DXGI Factory. If it is stale we need to create a new factory.
+            CreateFactory();
+        }
+
+        // Get the retangle bounds of the app window
+        RECT windowBounds;
+        GetWindowRect(m_window, &windowBounds);
+        const int ax1 = windowBounds.left;
+        const int ay1 = windowBounds.top;
+        const int ax2 = windowBounds.right;
+        const int ay2 = windowBounds.bottom;
+
+        ComPtr<IDXGIOutput> bestOutput;
+        int bestIntersectArea = -1;
+
+        ComPtr<IDXGIAdapter> adapter;
+        for (UINT adapterIndex = 0; SUCCEEDED(m_dxgiFactory->EnumAdapters(adapterIndex, adapter.ReleaseAndGetAddressOf())); adapterIndex++)
+        {
+            ComPtr<IDXGIOutput> output;
+            for (UINT outputIndex = 0; SUCCEEDED(adapter->EnumOutputs(outputIndex, output.ReleaseAndGetAddressOf())); outputIndex++)
+            {
+                // Get the rectangle bounds of current output
+                DXGI_OUTPUT_DESC desc;
+                ThrowIfFailed(output->GetDesc(&desc));
+                RECT r = desc.DesktopCoordinates;
```

Comment:
For clang tidiness sake, the parameters to ``ComputeIntersectionArea``, it's return value, and all these locals should be ``long`` instead of ``int`` since the underlying type of ``RECT`` members is actually ``long``. MSVC considers them the same, but they are not strictly speaking the same type.